zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/Interpreter.java

```diff
@@ -296,6 +302,40 @@ public Interpreter getInterpreterInTheSameSessionByClassName(String className) {
     return null;
   }
 
+  /**
+   * Replace markers #{contextFieldName} by values from {@link InterpreterContext} fields
+   * with same name and marker #{user}. If value == null then replace by empty string.
+   */
+  private void replaceContextParameters(Properties properties) {
+    InterpreterContext interpreterContext = InterpreterContext.get();
+    if (interpreterContext != null) {
+      String markerTemplate = "#\\{%s\\}";
+      List<String> skipFields = Arrays.asList("paragraphTitle", "paragraphId", "paragraphText");
+      List typesToProcess = Arrays.asList(String.class, Double.class, Float.class, Short.class,
+          Byte.class, Character.class, Boolean.class, Integer.class, Long.class);
+      for (String key : properties.stringPropertyNames()) {
+        String p = properties.getProperty(key);
+        if (StringUtils.isNotEmpty(p)) {
+          for (Field field : InterpreterContext.class.getDeclaredFields()) {
+            Class clazz = field.getType();
+            if (!skipFields.contains(field.getName()) && (typesToProcess.contains(clazz)
+                || clazz.isPrimitive())) {
+              try {
+                Object value = FieldUtils.readField(field, interpreterContext, true);
+                p = p.replaceAll(String.format(markerTemplate, field.getName()),
+                    value != null ? value.toString() : StringUtils.EMPTY);
+              } catch (Exception e) {
+                logger.error("Cannot replace context parameter", e);
```

Comment:
test testPropertyWithReplacedContextFields  covers this case, `replName` in context is NULL and replaced with an empty string